training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,137 @@
+                try {
+                    throw new UserNotFoundException();
+                } catch (UserNotFoundException e) {
+                    e.printStackTrace();
+                }
+                break;
+            case "bookIdMismatchException": {
+                try {
+                    throw new BookIdMismatchException();
+                } catch (BookIdMismatchException e) {
+                    e.printStackTrace();
+                }
+            }
+            break;
+            case "bookAlreadyOwnedException": {
+                try {
+                    throw new BookAlreadyOwnedException();
+                } catch (BookAlreadyOwnedException e) {
+                    e.printStackTrace();
+                }
+            }
+            break;
+
+            case "bookNotFoundException": {
+                try {
+                    throw new BookNotFoundException();
+                } catch (BookNotFoundException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    @DeleteMapping("/users/{id}")
+    public void delete(@PathVariable Long id) {
+        Optional<Users> userToDelete = userRepository.findById(id);
+        if (userToDelete.isPresent()) {
+            userToDelete.get().getBooks().removeAll(userToDelete.get().getBooks());
+            userRepository.delete(userToDelete.get());
+        } else {
+            this.generateException("userNotFoundException");
+        }
+    }
+
+
+    @PostMapping("/createUser")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Users create(@RequestBody Users users) {
+        Users newUsers = new Users(users.getUsername(), users.getName(), users.getBirthdate());
+        return userRepository.save(newUsers);
+    }
+
+    @PutMapping("/users/{id}")
+    public Users updateUser(@RequestBody Users users, @PathVariable Long id) {
+        if (users.getId() != id) {
+            this.generateException("userNotFoundException");
+        }
+        userRepository.findById(id)
+            .orElseThrow(UserNotFoundException::new);
+        return userRepository.save(users);
+    }
+
+    @PutMapping("/users/{id_user}/book/{id_book}")
+    public Users updateBookOfUser(@PathVariable String id_user, @PathVariable String id_book) {
+        Optional<Users> user = userRepository.findById(Long.parseLong(id_user));
+        if (user.isPresent()) {
+            Optional<Book> bookToAssing = bookRepository.findById(Long.parseLong(id_book));
+            if (bookToAssing.isPresent()) {
+                if (user.get().getBooks().stream()
+                    .anyMatch(book -> book.getId() == bookToAssing.get().getId())) {
+                    this.generateException("bookAlreadyOwnedException");
+                } else {
+                    user.get().addBook(bookToAssing.get());
+                    bookToAssing.get().addUser(user.get());
+                    bookRepository.save(bookToAssing.get());
```

Comment:
As this is a `many to many` relationship, you can just save the user and it will also update the users list of the book.